### DISCUSSION TITLE
Very Jerky jog movements since updating to UGS V2.1.7 Platform!

### DISCUSSION THREAD

**Discussion by LorneRanger** [winder/Universal-G-Code-Sender#2564] [General]
Hello Everyone,

New member first post. Have been noticing some erratic jog movements since updating from UGS V2.1.6 to 2.1.7 Platform. This is happening on a Windows 10 and Windows 11 based system. Very jerky movements at the beginning of a move when using the jog controller. Same thing with my Logitech game controller but not as prominent because you can ease into the move. Anyone else noticed this?

Lorne 😊

**Comment by breiler** [comment]
I have not seen this on my machines.

Can you do a jog where this happens, and copy'n'paste the console log messages to show what commands have been sent.

Can you also attach your UGS log file, it will give clues about your host computer specifications and any error messages.

What type of CNC controller and firmware are you using?

**Reply by breiler** [reply]
Forgot, the easiest way to get the log file is through the menu `Tools -> Open log directory`. The file is called `messages.log`

**Comment by LorneRanger** [comment]
Thanks for your reply . . . I'll get something for you soon and get back! One of the controllers I'm using is an Arduino Uno based stepper motor controller from Sienci Labs (grbl 1.1h). The other is a Makerbase MKS Gen L V1.0 (grbl 1.1t). This just started after installing V2.1.7. When I install an update I just run the latest update over the old one. Not sure if that's the proper procedure but have always done it this way.

**Comment by LorneRanger** [comment]
Grbl 1.1h ['$' for help] LongMill build Sept 8, 2021 

>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok
>>> $J=G21G91X0.367Y-0.367F3111.27
ok

Log File:
Hope I did this right?
[messages.log](https://github.com/user-attachments/files/16070946/messages.log)

**Reply by breiler** [reply]
> Hope I did this right?

Yep, the log looked fine. No errors or memory problems.

> When I install an update I just run the latest update over the old one

Don't copy folder on top of the other (merging them), there is a chance that there are old library files present that would be used instead of the new ones. From the log it looked correct - that they are in separate folders.

The jog commands are using a quite fast jog rate. Can your machine move that fast? What is your max rate configuration ($110 - $113)?

It could be that it generates a jog command that is expected to complete in ~0.007 seconds (0.36 / 3111 * 60). But if your max feed rate is ~1000 mm/min it would complete in ~0.02 seconds.

**Reply by LorneRanger** [reply]
Seeing some interaction happening when the Probe Module Plugin is launched. If it’s on and I press and hold the Jog Controller buttons for a continuous move you can see some jerking, stuttering happening just before it starts moving smoothly. When I close the Probe Module Plugin then try the same continuous move it works smooth right from the beginning and until you release the button. Are you able to replicate this on your machine?

When I previously said it seemed to be working fine on my Windows 7 laptop the Probe Module wasn’t on.

**Comment by LorneRanger** [comment]
Thanks for your help! 

These are my settings:
$100 = 160.050    (X-axis travel resolution, step/mm)
$101 = 160.000    (Y-axis travel resolution, step/mm)
$102 = 160.000    (Z-axis travel resolution, step/mm)
$110 = 3000.000    (X-axis maximum rate, mm/min)
$111 = 3000.000    (Y-axis maximum rate, mm/min)
$112 = 1500.000    (Z-axis maximum rate, mm/min)
$120 = 600.000    (X-axis acceleration, mm/sec^2)
$121 = 600.000    (Y-axis acceleration, mm/sec^2)
$122 = 600.000    (Z-axis acceleration, mm/sec^2)

I haven't had any problems running these settings in the past. Don't cut anywhere near these speeds, only for rapids.

Funny . . . Just hooked up my machine to the laptop I was previously using (Windows 7 Dell Inspiron 15R - I5) and it's working very smooth and not seeing the same jerky erratic issues as with the Windows 11 desktop computer UGS was hooked up to. Tells me it must be something with windows? Any ideas? Would AV software have anything to do with it?

What is the procedure for installing a new UGS update then that I should be following, Thanks!
